### PR TITLE
refactor: move monkey patchable portions of Knex to Knex package

### PR DESCRIPTION
**Pull request by thekevinbrown**
Moves the portions of Knex that are being monkey patched from dynamic requires to hard requires so webpack can trace them. They're optional, but each driver insists that they're received.

### CODE REVIEW

**Comment by B4nan** [comment]
@merceyz @arcanis will this work with yarn pnp?

**Comment by merceyz** [comment]
Yes, it will work perfectly fine. I don't quite see the point of the try catch though as the files will exist, which also makes the null checks unnecessary.

**Comment by B4nan** [comment]
Agreed, the whole knex will always be there, so it is safe to require all of it

**Comment by B4nan** [comment]
should be even possible to use `import` instead of `require`, right?

**Comment by thekevinbrown** [comment]
I wasn't sure if importing those knex files would also in turn hard require an optional dependency that users don't always have installed. For example if I'm using Postgres, and I require `'knex/lib/dialects/mysql'`, does that throw if I don't have the mysql package installed? If that's not the case then we don't need the try catches, I thought it might cause problems.

I'm wasn't sure about using `import` because it returns a promise, and we can't await before exporting, so we'd need to change the drivers to await the actual import. Can make that change if that's preferable.

**Comment by B4nan** [comment]
> I wasn't sure if importing those knex files would also in turn hard require an optional dependency that users don't always have installed. For example if I'm using Postgres, and I require 'knex/lib/dialects/mysql', does that throw if I don't have the mysql package installed? If that's not the case then we don't need the try catches, I thought it might cause problems.

Looking at the source code, requiring just the dialect won't load the connector, so it should be safe. 

> I'm wasn't sure about using import because it returns a promise, and we can't await before exporting, so we'd need to change the drivers to await the actual import. Can make that change if that's preferable.

I mean imports like `import * as ... from '...'`

**Comment by thekevinbrown** [comment]
Ah yup, totally can. I’ll make those changes tomorrow.

**Comment by thekevinbrown** [comment]
@B4nan, all done, let me know if there's anything else on this one that needs doing.

File: packages/sqlite/src/SqliteConnection.ts [hunk 1]
```diff
@@ -56,16 +54,21 @@ export class SqliteConnection extends AbstractSqlConnection {
    * monkey patch knex' sqlite Dialect so it returns inserted id when doing raw insert query
    */
   private getPatchedDialect() {
-    const processResponse = Dialect.prototype.processResponse;
-    Dialect.prototype.processResponse = (obj: any, runner: any) => {
+    const { Sqlite3Dialect } = MonkeyPatchable;
+    if (!Sqlite3Dialect) {
```
Comment (merceyz) [inline packages/sqlite/src/SqliteConnection.ts:58]: These checks shouldn't be needed as the object is guaranteed to exist

Comment (thekevinbrown) [reply packages/sqlite/src/SqliteConnection.ts:58]: Yup, my bad, I'll fix that.

Comment (thekevinbrown) [reply packages/sqlite/src/SqliteConnection.ts]: Should be good to go now. Thanks!

**Comment by thekevinbrown** [comment]
Actually, this isn't working for me: "Cannot destructure property 'PostgresDialectTableCompiler' of 'knex_1.MonkeyPatchable' as it is undefined."

Let me keep working and see what I can come up with.

**Comment by B4nan** [comment]
Looking good, we just need to cover that one last branch in the `Utils.ts`, as apparently now we never call `Utils.requireFrom()` with a path to directory in the second param (I'd vote for adding simple unit test for that method into `Utils.test.ts` instead of refactoring the method to the minimum).

**Comment by merceyz** [comment]
Just a note, when you force push it makes it really hard to see what changed so we need to go over all of the code again, the PR will be squash merged so you don't have to worry about keeping one commit

**Comment by thekevinbrown** [comment]
How does that work with the angular commit messages?

**Comment by B4nan** [comment]
There will be only one commit in the master (after merging), and that is what will be used for changelog.

**Comment by B4nan** [comment]
Thanks again! And don't forget to let me know if we need to adjust the webpack docs (if there are changes needed in the config mainly).

**Pull request merged**